packages/circuit-ui/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,267 @@
+`;
+
+const Overlay = styled.div(
+  ({ theme }: StyleProps) => css`
+    ${theme.mq.untilKilo} {
+      position: fixed;
+      top: 0;
+      bottom: 0;
+      left: 0;
+      right: 0;
+      background-color: ${theme.colors.overlay};
+    }
+  `,
+);
+
+type Divider = { type: 'divider' };
+type Action = PopoverItemProps | Divider;
+
+function isDivider(action: Action): action is Divider {
+  return 'type' in action && action.type === 'divider';
+}
+
+export interface PopoverProps {
+  isOpen: boolean;
+  actions: Action[];
+  referenceElement: RefObject<HTMLElement>;
+  placement?: Placement;
+  onClose: (event: Event) => void;
+}
+
+export const Popover = ({
+  isOpen,
+  onClose,
+  actions,
+  referenceElement,
+  placement = 'bottom',
+  ...props
+}: PopoverProps): JSX.Element | null => {
+  const theme: Theme = useTheme();
+
+  // Popper custom modifier to apply bottom sheet for mobile.
+  // The window.matchMedia() is a useful API for this, it allows you to change the styles based on a condition.
+  // useMemo hook is used in order to prevent the render loop, more https://popper.js.org/react-popper/v2/faq/#why-i-get-render-loop-whenever-i-put-a-function-inside-the-popper-configuration
+  const mobilePosition: Modifier<'mobilePosition', { state: State }> = useMemo(
+    () => ({
+      name: 'mobilePosition',
+      enabled: true,
+      phase: 'write',
+      fn({ state }) {
+        if (window.matchMedia(`${theme.breakpoints.untilKilo}`).matches) {
+          // eslint-disable-next-line no-param-reassign
+          state.styles.popper = {
+            width: '100%',
+            left: '0px',
+            right: '0px',
+            bottom: '0px',
+            position: 'fixed',
+          };
+        } else {
+          // eslint-disable-next-line no-param-reassign
+          state.styles.popper.width = 'auto';
+        }
+      },
+    }),
+    [theme],
+  );
+  // Note: the usePopper hook intentionally takes the DOM node, not refs, in order to be able to update when the nodes change.
+  // A callback ref is used here to permit this behaviour, and useState is an appropriate way to implement this.
+  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
+  const { styles, attributes } = usePopper(
+    referenceElement.current,
+    popperElement,
+    {
+      placement,
+      modifiers: [
+        mobilePosition,
+        // The flip modifier is used if the popper has placement set to bottom, but there isn't enough space to position the popper in that direction.
+        // By default, it will change the popper placement to top. As soon as enough space is detected, the placement will be reverted to the originally defined (or preferred) one.
+        // You can also define fallback placements by providing a list of placements to try. When no space is available on the preferred placement, the modifier will test the ones provided in the list, and use the first useful one.
+        {
+          name: 'flip',
```

Comment:
Also: I would move `flip` outside of the `usePopper` for readability/consistency with `mobilePosition` 🙂